packages/flutter_tools/lib/src/protocol_discovery.dart

```diff
@@ -21,37 +21,37 @@ class ProtocolDiscovery {
   final DeviceLogReader _logReader;
   final String _serviceName;
 
-  Completer<int> _completer = new Completer<int>();
+  Completer<Uri> _completer = new Completer<Uri>();
   StreamSubscription<String> _subscription;
 
   /// The [Future] returned by this function will complete when the next service
-  /// protocol port is found.
-  Future<int> nextPort() => _completer.future;
+  /// Uri is found.
+  Future<Uri> nextUri() => _completer.future;
```

Comment:
Per our discussion, the name `nextUri` is more indicative of what this method does, especially given the fact that the class is named `ProtocolDiscovery`.